vm/src/obj/objlist.rs

```diff
@@ -468,25 +470,31 @@ impl PyList {
     }
 
     #[pymethod(name = "__mul__")]
-    fn mul(&self, counter: isize, vm: &VirtualMachine) -> PyObjectRef {
-        let new_elements = sequence::seq_mul(&self.borrow_sequence(), counter)
-            .cloned()
-            .collect();
-        vm.ctx.new_list(new_elements)
+    fn mul(&self, counter: isize, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        if counter < 0 || self.len() * (counter as usize) < MAX_MEMORY_SIZE {
+            let new_elements = sequence::seq_mul(&self.borrow_sequence(), counter)
+                .cloned()
+                .collect();
+            return Ok(vm.ctx.new_list(new_elements));
+        }
+        return Err(vm.new_memory_error("".to_owned()));
     }
 
     #[pymethod(name = "__rmul__")]
-    fn rmul(&self, counter: isize, vm: &VirtualMachine) -> PyObjectRef {
+    fn rmul(&self, counter: isize, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
         self.mul(counter, &vm)
     }
 
     #[pymethod(name = "__imul__")]
-    fn imul(zelf: PyRef<Self>, counter: isize) -> PyRef<Self> {
-        let new_elements = sequence::seq_mul(&zelf.borrow_sequence(), counter)
-            .cloned()
-            .collect();
-        zelf.elements.replace(new_elements);
-        zelf
+    fn imul(zelf: PyRef<Self>, counter: isize) -> PyResult<Self> {
```

Comment:
I'm not sure how to handle this function, since it also needs to throw a MemoryError.

Reply:
You can add a `vm: &VirtualMachine` parameter at the end of the signature to get the `vm`.